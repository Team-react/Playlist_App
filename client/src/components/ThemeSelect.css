.container{
  width: auto;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  transition: 2s linear;
}

.authdiv{
  margin-top: 10%;
  /* background-color: rgba(255, 255, 255, 0.5); */
  border-radius: 20px;
  height: 280px;
  width: 400px;
}

.textbox{
  font-size: 25px;
  margin-bottom: 10px;
  width: 350px;
  height: 70px;
  border-radius: 11px;
  text-align: center;
}

.btnpos{
  margin-left: 34%;
}

.readybtn{
  height: 50px;
  width: 100px;
  border-radius: 10px;
  background-color: rgb(52, 163, 8);
  font-weight: bold;
  color: white;
  font-size: 15px;
  float: left;
  outline: 0;
}

.readybtn:disabled{
  height: 50px;
  width: 100px;
  border-radius: 10px;
  background-color: rgb(100, 100, 100);
  font-weight: bold;
  color: white;
  font-size: 15px;
  float: left;
  outline: 0;
  opacity: 0.5;
}

/* .readybtn:hover{
  height: 50px;
  width: 100px;
  border-radius: 10px;
  background-color: rgba(51, 163, 8, 0.5);
  font-weight: bold;
  color: white;
} */