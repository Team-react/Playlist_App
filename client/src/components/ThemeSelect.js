import React, { Component } from "react";
import SpotifyWebApi from 'spotify-web-api-node';
import PlaylistGenerator from './PlaylistGenerator'


var spotifyApi = new SpotifyWebApi();


 
class ThemeSelect extends Component {
  constructor(props){
    super(props);
    this.state = {
      desiredDuration: 0,
      playlist_type: '',
      playlistComplete: false,
      renderChild: false
    }
    this.handleChildUnmount = this.handleChildUnmount.bind(this);

  }
  handleChildUnmount(){
    this.setState({renderChild: false});
}
handleChildmount(){
  this.setState({renderChild: true});

}
    
  getHashParams() {
    var hashParams = {};
    var e, r = /([^&;=]+)=?([^&;]*)/g,
        q = window.location.hash.substring(1);
    e = r.exec(q)
    while (e) {
       hashParams[e[1]] = decodeURIComponent(e[2]);
       e = r.exec(q);
    }
    return hashParams;
  }

  setToken() {
    console.log(this.props.token)
    spotifyApi.setAccessToken(this.props.token)

  }

  playlistIsComplete(){
    this.setState({
      playlistComplete: true

    })
  }
  playlistIsNotComplete(){
    this.setState({
      playlistComplete: false
    })
  }
  
  // getRandomPlaylist(genre) {

  //   spotifyApi.setAccessToken(this.props.token)

    

  //   spotifyApi.searchPlaylists(genre)
  //   .then((data) => {
  //     console.log(data, 'its not even reach this point')
  //     var numberOfPlaylists = (data.body.playlists.items).length
  //     console.log(numberOfPlaylists)
  //     this.props.playlist(data.body.playlists.items[Math.floor(Math.random() * numberOfPlaylists)].id)
  //   }, function(err) {
  //     console.log('Something went wrong!', err);
  //   });
  // }
  changeHandler = event => {
    var time = parseInt(event.target.value) * 60000
    console.log(time)

    this.setState({
      desiredDuration: time
    });
  }

  // playlistHandler = event => {
  //   console.log(this.state.playlist_type)

  //   event.preventDefault();

  //   this.getRandomPlaylist(this.state.playlist_type)
  // }

  playlistTypeHandler = event => {
    this.setState({
      playlist_type: event.target.value

    })
    this.handleChildmount()
  }

  render() {
    return (
      <>
      <div>

      {this.state.renderChild ?

      <PlaylistGenerator
      playListType={this.state.playlist_type}
      desiredDuration={this.state.desiredDuration}
      token={this.props.token}
      playlistIsComplete={this.playlistIsComplete.bind(this)}
      playlistIsNotComplete={this.playlistIsNotComplete.bind(this)}
      playlistComplete={this.state.playlistComplete}
      unmountMe={this.handleChildUnmount.bind(this)}
      
      />
      : null
      }
      </div>
      
      <div>{ !(this.state.playlistComplete) &&
      <div>
      <form>
      <input id='input' type="text" name="playlist_type" 
      placeholder="Input artist or genre" 
      onChange={this.playlistTypeHandler} 
      />
    

    </form>
      <form>
      <input type="text"
             name="duration"
             placeholder="input playlist length"
            //  value={this.state.desiredDuration}
             onChange={this.changeHandler}
      />
    </form>
    </div>
  }</div>

    
 
    </>
    )
  }
}
 
export default ThemeSelect;