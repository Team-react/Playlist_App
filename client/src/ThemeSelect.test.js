import React from 'react';
import { shallow } from 'enzyme';
import ThemeSelect from './components/ThemeSelect';
describe('ThemeSelect', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ThemeSelect debug />);
  
    expect(component).toMatchSnapshot();
  });
});