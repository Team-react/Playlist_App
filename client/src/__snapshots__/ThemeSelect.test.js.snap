// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSelect should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
